CDP4Common/Poco/FileRevision.cs

```diff
@@ -40,14 +40,15 @@ protected string GetDerivedPath()
         {
             var path = new StringBuilder();
             var containingFolder = this.ContainingFolder;
+
             if (containingFolder != null)
             {
                 path.Append(containingFolder.Path);
                 path.Append("/");
+                path.Append(this.ContainingFolder.Name);
+                path.Append("/");
```

Comment:
The SDK gets used on both server and client machines with potentially different OSs. I dont know whether it is relevant for this method as I dont know where it is being used but perhaps you can use https://docs.microsoft.com/en-us/dotnet/api/system.io.path.directoryseparatorchar?view=netframework-4.8 for this character?

Reply:
This is not being user server-side. We don't store the files on the filesystem on the server using the "virtual filesystem" that users can create using the IME. The files are stored in folders based on the SHA-1 hash of the file, and we store each file only 1 time, even though it may exist multiple times in the virtual filestore(s).

regarding directory separator character, here are the docs of the .path property of the Folder class:

```
Note: The path is derived to be the concatenation of the folder names in the hierarchy specified through containingFolder. This yields a path that is similar to that of "file://" URL starting from the containing FileStore.
```
note the use of forward slash there, that is why it has been implemented in that way.